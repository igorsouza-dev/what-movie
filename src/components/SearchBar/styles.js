import styled from 'styled-components';

export const Container = styled.form`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin-right: 10px;
`;

export const Input = styled.input`
  width: 100%;
  max-width: 300px;
  border: 0;
  border-radius: 4px 0 0 4px;
  background: #eee;
  padding: 5px 10px;
  height: ${props => (props.small ? '30px' : '40px')};
  color: #333;
  font-weight: bold;
  &::placeholder {
    color: #bbb;
    font-weight: bold;
    font-size: 14px;
  }
`;
export const Button = styled.button`
  width: 60px;
  border: 0;
  border-radius: 0 4px 4px 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #d57d56;
  height: ${props => (props.small ? '30px' : '40px')};
`;
