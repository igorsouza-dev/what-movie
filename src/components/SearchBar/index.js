import React, { useState } from 'react';
import { FaSearch } from 'react-icons/fa';
import { Container, Input, Button } from './styles';

export default function SearchBar({ onSearch }) {
  const [input, setInput] = useState('');
  function handleSubmit(e) {
    e.preventDefault();
    onSearch(input);
  }
  return (
    <Container data-testid="search-bar" onSubmit={handleSubmit}>
      <Input
        type="text"
        placeholder="Type the name of a movie"
        value={input}
        onChange={e => setInput(e.target.value)}
      />
      <Button type="submit">
        <FaSearch color="#fff" size={24} />
      </Button>
    </Container>
  );
}
