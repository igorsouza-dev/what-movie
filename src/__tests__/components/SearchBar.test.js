import React from 'react';
import { render, cleanup } from '@testing-library/react';

import SearchBar from 'components/SearchBar';

describe('<SearchBar />', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container, getByTestId } = render(<SearchBar />);
    const linkElement = getByTestId(/search-bar/i);
    expect(linkElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
