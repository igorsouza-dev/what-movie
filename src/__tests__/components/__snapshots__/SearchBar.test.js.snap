// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should render 1`] = `
<div
  class="sc-AykKC cGRdXF"
  data-testid="search-bar"
>
  <input
    class="sc-AykKD dRqlXC"
    placeholder="Type the name of a movie"
    type="text"
  />
</div>
`;
